contributors/design-proposals/encryption.md

```diff
@@ -0,0 +1,204 @@
+# Encryption
```

Comment:
You raise some good questions, I do believe there's been some talk around this already in google docs. 

It's true that most clouds do support key management systems out of the box, hence why we want the encryption interface to be pluggable for those use cases. @smarterclayton may have stronger/better reasonings than I do on this but I had thought we want to support a builtin solution for this because:

1) We need to support this for bare metal setups
2) Using third party vendors like Vault would require support for the storage backends it depends on. 
3) Using third party tools like Vault means any pods that require Secrets heavily depend on the availability of those vendors. This adds operational overhead for cluster admins.

For 2) and 3) I can see Kubernetes supporting a highly available cluster add-on for Vault that would work in most setups (similar to kube-dns, not required but highly recommended and a highly available setup is provided by the kubernetes community). 

For 2) etcd is an optional storage backend for Vault so we may not have to worry about adding an additional storage backend, but there are some issues we should address if we go down that path (e.g. Vault putting too much load on etcd servers will make master server unavailable). 

Would love to discuss this with you in more details, feel free to ping me on kubernetes slack to setup a time :)